apps/contrib/elasticsearch.py

```diff
@@ -25,20 +25,26 @@ def _get_es_body(self, for_count=False):
         return body
 
     def _get_results_from_hits(self, hits):
-        pks = [hit['fields']['pk'][0] for hit in hits]
-        scores = {str(hit['fields']['pk'][0]): hit['_score'] for hit in hits}
+        pks = []
+        scores = {}
         highlights = {}
 
-        results = {str(pk): None for pk in pks}
         for hit in hits:
             hs = hit['highlight']
             hs.pop('content_type')
+            # ignore hits from the _all_text field as it doesn't work with
+            # our translations
+            hs.pop('_all_text_boost_2_0')
             for highlight in hs.values():
                 # we only need one highlight
                 if hit['fields']['pk'][0] not in highlights:
                     highlights[str(hit['fields']['pk'][0])] = highlight[0]
+                    # only use search results which have a valid hit
+                    pks.append(hit['fields']['pk'])
```

Comment:
hard to imagine the data structure without an example and not inline documentation, but looking at the top, isn't here missing a `[0]`?

Reply:
good catch :facepalm:  there's also another mistake. Will push a fix

Reply:
I also added some general comments, but most of the code comes directly from wagtail anyway